Integration tests failing on Travis

Since a couple of days ago I've see the "Lookoutd Integration Tests Linux" fail on Travis. They usually work if you retry it manually.

One of the failing logs:
```

$ LOOKOUT_TEST_QUEUE=true make test-json
rm -rf build build/bin 
go clean .
go build -ldflags "-X main.version=dev-362f0e3 -X main.build=12-05-2018_11_25_07 -X main.commit=362f0e3" -o "build/bin/dummy" ./cmd/dummy
go build -ldflags "-X main.version=dev-362f0e3 -X main.build=12-05-2018_11_25_09 -X main.commit=362f0e3" -o "build/bin/lookoutd" ./cmd/lookoutd
go build -ldflags "-X main.version=dev-362f0e3 -X main.build=12-05-2018_11_25_12 -X main.commit=362f0e3" -o "build/bin/lookout-sdk" ./cmd/lookout-sdk
DUMMY_BIN=/home/travis/gopath/src/github.com/src-d/lookout/build/bin/dummy \
	LOOKOUT_BIN=/home/travis/gopath/src/github.com/src-d/lookout/build/bin/lookoutd \
	go test -v -parallel 1 -tags='integration bblfsh' github.com/src-d/lookout/cmd/server-test
=== RUN   TestDummyIntegrationSuite
=== RUN   TestDummyIntegrationSuite/TestReviewDontPostSameComment
running lookoutd migrate
starting lookoutd watch --provider json -c ../../fixtures/dummy_config.yml
starting lookoutd work --provider json -c ../../fixtures/dummy_config.yml --probes-addr 0.0.0.0:8091
starting dummy serve --files
stopping services
=== RUN   TestDummyIntegrationSuite/TestSuccessPush
running lookoutd migrate
starting lookoutd watch --provider json -c ../../fixtures/dummy_config.yml
starting lookoutd work --provider json -c ../../fixtures/dummy_config.yml --probes-addr 0.0.0.0:8091
starting dummy serve --files
stopping services
=== RUN   TestDummyIntegrationSuite/TestSuccessReview
running lookoutd migrate
starting lookoutd watch --provider json -c ../../fixtures/dummy_config.yml
starting lookoutd work --provider json -c ../../fixtures/dummy_config.yml --probes-addr 0.0.0.0:8091
starting dummy serve --files
stopping services
=== RUN   TestDummyIntegrationSuite/TestWrongRevision
running lookoutd migrate
starting lookoutd watch --provider json -c ../../fixtures/dummy_config.yml
starting lookoutd work --provider json -c ../../fixtures/dummy_config.yml --probes-addr 0.0.0.0:8091
starting dummy serve --files
stopping services
--- PASS: TestDummyIntegrationSuite (26.50s)
    --- PASS: TestDummyIntegrationSuite/TestReviewDontPostSameComment (6.94s)
    --- PASS: TestDummyIntegrationSuite/TestSuccessPush (6.64s)
    --- PASS: TestDummyIntegrationSuite/TestSuccessReview (6.51s)
    --- PASS: TestDummyIntegrationSuite/TestWrongRevision (6.41s)
=== RUN   TestErrorAnalyzerIntegrationSuite
=== RUN   TestErrorAnalyzerIntegrationSuite/TestAnalyzerErr
running lookoutd migrate
starting lookoutd watch --provider json -c ../../fixtures/dummy_config.yml
starting lookoutd work --provider json -c ../../fixtures/dummy_config.yml --probes-addr 0.0.0.0:8091
stopping services
--- PASS: TestErrorAnalyzerIntegrationSuite (6.42s)
    --- PASS: TestErrorAnalyzerIntegrationSuite/TestAnalyzerErr (6.42s)
=== RUN   TestMultiDummyIntegrationSuite
=== RUN   TestMultiDummyIntegrationSuite/TestSuccessReview
running lookoutd migrate
starting lookoutd watch --provider json -c ../../fixtures/double_dummy_config.yml
starting lookoutd work --provider json -c ../../fixtures/double_dummy_config.yml --probes-addr 0.0.0.0:8091
starting dummy serve --files
starting dummy serve --files --analyzer ipv4://localhost:10303
stopping services
--- PASS: TestMultiDummyIntegrationSuite (8.38s)
    --- PASS: TestMultiDummyIntegrationSuite/TestSuccessReview (8.38s)
=== RUN   TestProxyIntegrationSuite
=== RUN   TestProxyIntegrationSuite/TestParseOk
starting lookoutd watch --provider json -c ../../fixtures/empty_config.yml
starting lookoutd work --provider json -c ../../fixtures/empty_config.yml --probes-addr 0.0.0.0:8091
stopping services
--- FAIL: TestProxyIntegrationSuite (5.08s)
    --- FAIL: TestProxyIntegrationSuite/TestParseOk (5.08s)
        proxy_test.go:55: 
            	Error Trace:	proxy_test.go:55
            	Error:      	Received unexpected error:
            	            	rpc error: code = Unavailable desc = all SubConns are in TransientFailure, latest connection error: connection error: desc = "transport: Error while dialing dial tcp 127.0.0.1:10301: connect: connection refused"
            	Test:       	TestProxyIntegrationSuite/TestParseOk
        proxy_test.go:56: 
            	Error Trace:	proxy_test.go:56
            	Error:      	Expected value not to be nil.
            	Test:       	TestProxyIntegrationSuite/TestParseOk
FAIL
FAIL	github.com/src-d/lookout/cmd/server-test	46.398s
make: *** [test-json] Error 1
The command "LOOKOUT_TEST_QUEUE=true make test-json" exited with 2.

```

--- a/cmd/server-test/dummy_analyzer_test.go
+++ b/cmd/server-test/dummy_analyzer_test.go
@@ -24,7 +24,7 @@ func (suite *DummyIntegrationSuite) SetupTest() {
 	suite.r, suite.w = suite.StartLookoutd(dummyConfigFile)
 
 	suite.StartDummy("--files")
-	suite.GrepTrue(suite.r, `connection state changed to 'READY'`)
+	suite.GrepTrue(suite.r, `msg="connection state changed to 'READY'" addr="ipv4://localhost:9930" analyzer=Dummy`)
 }
 
 func (suite *DummyIntegrationSuite) TearDownTest() {

--- a/cmd/server-test/error_analyzer_test.go
+++ b/cmd/server-test/error_analyzer_test.go
@@ -59,7 +59,7 @@ func (suite *ErrorAnalyzerIntegrationSuite) SetupTest() {
 	suite.r, suite.w = suite.StartLookoutd(dummyConfigFile)
 
 	suite.startAnalyzer(suite.Ctx, &errAnalyzer{})
-	suite.GrepTrue(suite.r, `connection state changed to 'READY'`)
+	suite.GrepTrue(suite.r, `msg="connection state changed to 'READY'" addr="ipv4://localhost:9930" analyzer=Dummy`)
 }
 
 func (suite *ErrorAnalyzerIntegrationSuite) TearDownTest() {

--- a/cmd/server-test/multi_analyzer_test.go
+++ b/cmd/server-test/multi_analyzer_test.go
@@ -22,10 +22,10 @@ func (suite *MultiDummyIntegrationSuite) SetupTest() {
 	suite.r, suite.w = suite.StartLookoutd(doubleDummyConfigFile)
 
 	suite.StartDummy("--files")
-	suite.GrepTrue(suite.r, `connection state changed to 'READY'`)
+	suite.GrepTrue(suite.r, `msg="connection state changed to 'READY'" addr="ipv4://localhost:9930" analyzer=Dummy1`)
 
 	suite.StartDummy("--files", "--analyzer", "ipv4://localhost:10303")
-	suite.GrepTrue(suite.r, `connection state changed to 'READY'`)
+	suite.GrepTrue(suite.r, `msg="connection state changed to 'READY'" addr="ipv4://localhost:10303" analyzer=Dummy2`)
 }
 
 func (suite *MultiDummyIntegrationSuite) TearDownTest() {

--- a/cmd/server-test/proxy_test.go
+++ b/cmd/server-test/proxy_test.go
@@ -5,6 +5,7 @@ package server_test
 
 import (
 	"context"
+	"strings"
 	"testing"
 	"time"
 
@@ -27,6 +28,12 @@ func (suite *ProxyIntegrationSuite) SetupTest() {
 
 	suite.StoppableCtx()
 	suite.r, suite.w = suite.StartLookoutd(emptyConfigFile)
+
+	// Proxy can connect before or after the "Starting watcher" message is found
+	msg := `msg="connection state changed to 'READY'" addr="localhost:9432" app=lookoutd name=bblfsh-proxy`
+	if !strings.Contains(suite.Output(), msg) {
+		suite.GrepTrue(suite.r, msg)
+	}
 }
 
 func (suite *ProxyIntegrationSuite) TearDownTest() {

--- a/util/cmdtest/cmds.go
+++ b/util/cmdtest/cmds.go
@@ -241,3 +241,8 @@ func (suite *IntegrationSuite) runQuery(db *sql.DB, query string) {
 	_, err := db.Exec(query)
 	suite.Require().NoError(err, "can't execute SQL: %q", query)
 }
+
+// Output returns the output read so far from the reader returned in StartLookoutd
+func (suite *IntegrationSuite) Output() string {
+	return suite.logBuf.String()
+}

--- a/util/grpchelper/helper.go
+++ b/util/grpchelper/helper.go
@@ -32,17 +32,23 @@ func DialContext(ctx context.Context, target string, opts ...grpc.DialOption) (*
 	return pb.DialContext(ctx, target, opts...)
 }
 
+func logConnStatus(l log.Logger, state connectivity.State) {
+	if state == connectivity.TransientFailure {
+		l.Warningf("connection failed")
+	} else {
+		l.Infof("connection state changed to '%s'", state)
+	}
+}
+
 // LogConnStatusChanges logs gRPC connection status changes
 func LogConnStatusChanges(ctx context.Context, l log.Logger, conn *grpc.ClientConn) {
 	state := conn.GetState()
+	logConnStatus(l, state)
+
 	for {
 		if conn.WaitForStateChange(ctx, state) {
 			state = conn.GetState()
-			if state == connectivity.TransientFailure {
-				l.Warningf("connection failed")
-			} else {
-				l.Infof("connection state changed to '%s'", state)
-			}
+			logConnStatus(l, state)
 		} else {
 			// ctx expired / canceled, stop listing
 			return

--- a/util/grpchelper/proxy.go
+++ b/util/grpchelper/proxy.go
@@ -8,6 +8,7 @@ import (
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/metadata"
+	log "gopkg.in/src-d/go-log.v1"
 )
 
 // NewBblfshProxyServer creates gRPC server that proxies call to bblfsh
@@ -21,6 +22,11 @@ func NewBblfshProxyServer(addr string) (*grpc.Server, error) {
 	if err != nil {
 		return nil, err
 	}
+
+	go LogConnStatusChanges(context.Background(), log.DefaultLogger.With(log.Fields{
+		"name": "bblfsh-proxy",
+		"addr": addr,
+	}), bblfshConn)
 
 	// bblfsh proxy director
 	director := func(ctx context.Context, fullMethodName string) (context.Context, *grpc.ClientConn, error) {

